app/controllers/OrderController.java

```diff
@@ -37,24 +46,15 @@ public Result show() {
     }
 
     public Result submit() {
-        try {
-            final Cart currentCart = cartService.getOrCreateCart(session());
-            final Cart clearedCart = cartService.clearCart(currentCart);
-            return ok(success.render());
-        } catch (Exception e) {
-            Logger.error(e.getMessage(), e);
-            return internalServerError();
-        }
+        LOG.debug("Submitting Order details page");
+        final Cart currentCart = cartService.getOrCreateCart(session());
+        cartService.clearCart(currentCart);
```

Comment:
That also confused me the last time, the method implementation looks like it loads the cart to just flush it. Maybe the name of the method is not matching.